modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -799,6 +791,12 @@ in {
           force = true;
         };
     }));
+
+    # Mimic nixpkgs package environment for read-only profiles.ini management
+    home.sessionVariables = {
+      MOZ_LEGACY_PROFILES = 1;
+      MOZ_ALLOW_DOWNGRADE = 1;
```

Comment:
I'm not entirely certain this is safe since it will apply across the entire user session. E.g., it would presumably also apply for a secondary Firefox install that is not managed through Home Manager. Which perhaps causes unexpected behavior.

Setting these variables in a wrapper like in the main Nixpkgs firefox package seems safer.

Reply:
That may be correct for nixpkgs derivation (though there is no wrapped flavor for Darwin, at the moment at least). For homebrew installations that a lot of people use (nixpkgs ports for Darwin are very much new and still not complete - like the wrapper part that is missing), nixpkgs solution wouldn't help.